posts/xavdid-does-12in23.md

```diff
@@ -0,0 +1,254 @@
+> this post originally appeared on [David's website](https://xavd.id/blog/post/12-languages-in-12-months/) and is republished here with permission
+
+Last January, Exercism [announced](https://forum.exercism.org/t/the-12in23-challenge/2213) a new program called `12in23`, where they challenged participants to try 12 new programming languages in 2023. Each month would have a theme (such as "Analytical April" or "Object Oriented October") and would highlight specific languages to try. I love learning new things and have become a bit of a (programming) language nerd, so I decided to give it a try. 12 languages, 12 months!
+
+Now that the year is almost over I'm incredibly pleased with how the project went. I successfully tried 12 new languages, met great people in the Exercism community, and shipped some cool open source contributions along the way! In this post, I run through them all and talk about what I got from each.
+
+## Picking Languages
+
+I laid out a few guidelines for the year to help me make the most of the experience:
+
+1. Languages should be either brand new to me or at least unfamiliar enough that I'd feel like I was learning a lot.
+2. Selected languages should be (potentially) practical for me to learn more of in the future. This project was just for fun, but I want to be spending time learning (at least partially) useful things.
+3. I'd install all the local tooling & VSCode plugin for any language I was using. I wanted to compare languages on roughly equal footing, with as much type hinting and intellisense as possible. In college I did all my programming homework in Sublime Text without any linting or auto-complete. I worried that if I learned to program while using all those tools, I'd rely on them too much and wouldn't be a good programmer. Instead, the opposite has happened. The more cognition I can offload to my tools, the more I can think about the actual problem at hand. Don't remember things, [remember how to find them](https://hynek.me/articles/productive-fruit-fly-programmer/).
+
+Let's get to it!
+
+![](./images/badges.png) - FIXME
```

Comment:
These will either need to be removed or updated to point at your CDN

Reply:
If you drop the images into a seperate PR at https://github.com/exercism/images/tree/main/images (use a new folder called `community/` and prefix them with `xavdid-` (e.g. the path would be `REPO/images/community/xavdid-12in23-badges.png` or something). Then the URL would be `https://assets.exercism.org/images/community/xavdid-12in23-badges.png` :)

Reply:
https://github.com/exercism/images/pull/22